galactic: updated ci-job and codecov

JanStaschulat opened the pull request:
- updating the codecov link
- fixing issues regarding the CI workflow

JanStaschulat commented:
@pablogs9 , any idea, why the CI job for galactic is not started with the pull request?

pablogs9 commented:
@JanStaschulat I think that the workflows only are available if they are in the `default` branch: https://github.community/t/workflow-files-only-picked-up-from-master/16129

christophebedard commented:
I think if you change this from `master` to `galactic` (or just remove the line altogether) it should work:

https://github.com/ros2/rclc/blob/de019cf4bfd409ba5488d78ae2c19e1159ae9c83/.github/workflows/ci.yml#L7

JanStaschulat commented:
but it fails with the following message in the CI workflow:
```

Invoking "sudo rosdep init"
  /usr/bin/sudo rosdep init
  Wrote /etc/ros/rosdep/sources.list.d/20-default.list
  Recommended: please run
  
  	rosdep update
  
Error: Input has invalid distribution names.
```

christophebedard commented:
> but it fails with the following message in the CI workflow:
> 
> ```
> 
> Invoking "sudo rosdep init"
>   /usr/bin/sudo rosdep init
>   Wrote /etc/ros/rosdep/sources.list.d/20-default.list
>   Recommended: please run
>   
>   	rosdep update
>   
> Error: Input has invalid distribution names.
> ```

yeah, Galactic (release testing) isn't supported by `setup-ros` or `action-ros-ci` just yet. See https://github.com/ros-tooling/setup-ros/pull/387

christophebedard commented:
@JanStaschulat this should now be supported by `setup-ros` or `action-ros-ci`, although you'll need to use `@master` for now until we create a new release.

JanStaschulat commented:
> @JanStaschulat this should now be supported by `setup-ros` or `action-ros-ci`, although you'll need to use `@master` for now until we create a new release.

@christophebedard thanks for the information. I upgraded `setup-ros`and `action-ros-ci` to the most recent versions. But it still fails for the `Galactic` branch. I guess, I have to wait until next week when Galactic is officially released?

christophebedard commented:
If you use `@master`

```yaml
- uses: ros-tooling/setup-ros@master
...
- uses: ros-tooling/action-ros-ci@master
...
```

You can use `galactic` as the `ros_distribution` (with the `use-ros2-testing: true` option for `setup-ros` ).

christophebedard commented:
..and by upgrading to `@master` you'll need to replace

```yaml
      colcon-mixin-name: coverage-gcc
```

with

```yaml
      ​colcon-defaults​: ​|​
​        {​
​          "build": {​
​            "mixin": ["coverage-gcc"]​
​          }​
​        }​
```

😅

JanStaschulat commented:
For some reason, the CI jobs don't run.

christophebedard commented on .github/workflows/ci.yml:
```diff
@@ -25,19 +25,25 @@ jobs:
+          "build": {
+            "mixin": [ "coverage-gcc" ]
+          }
+        }
```
the indentation is wrong here. The 5 lines above should be indented by two spaces more.

christophebedard commented:
Galactic has been released and the actions have just been updated to fully support it, so you can now remove `use-ros2-testing: true` and use `ros-tooling/setup-ros@0.2.0` (or `@v0.2`) and `ros-tooling/action-ros-ci@0.2.1` (or `@v0.2`).

JanStaschulat commented:
@christophebedard Thanks for the notification. I updated the ci-job. Let's see if the bloom release works out. I updated also the branch version for rclc bloom-release in rosdistro: https://github.com/ros/rosdistro/pull/29683

JanStaschulat commented:
@christophebedard Do you have an idea, why the bloom realease `Gpr__rclc__ubuntu_focal_amd64` is not running?

christophebedard commented:
> @christophebedard Do you have an idea, why the bloom realease for Galactic `Gpr__rclc__ubuntu_focal_amd64` is not running?

I don't know, sorry. Looks like it's running for other/newer PRs though.

JanStaschulat commented:
@christophebedard i merged it without the test on the build farm. I created a new bloom release (see issue https://github.com/ros2/rclc/issues/96) for galactic.
